### PR TITLE
HDDS-7766. Extract Acl checks-related methods into a separate class

### PR DESCRIPTION
**Pull request by myskov**
## What changes were proposed in this pull request?

I moved all ACL check-related methods to AclManager to:
- make OzoneManager more readable
- make AclManager responsible for ACL-related actions
- decrease code duplication

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-7766

## How was this patch tested?

unit tests

### CODE REVIEW

**Comment by kerneltime** [comment]
@myskov can you please resolve conflicts and fix CI failures?

**Comment by kerneltime** [comment]
This looks like a good clean up, will need to check if this has any impact for ranger integration. @myskov please help get this PR into a good shape with the CI runs.

**Comment by smengcl** [comment]
Thanks @myskov for working on this.

Since there are already conflicts in `OzoneManager`, would you mind holding this off for ~7 days until snapshot feature branch is merged to master? After that I could help you on rebasing the patch and resolving the conflicts. Thanks.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/AclManager.java [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+
+    checkAllAcls(resType, store, acl,
+        vol, bucket, key,
+        user != null ? user : getRemoteUser(),
+        remoteIp != null ? remoteIp : omRpcAddress.getAddress(),
+        remoteIp != null ? remoteIp.getHostName()
+            : omRpcAddress.getHostName());
+  }
+
+  @SuppressWarnings("parameternumber")
```
Comment (neils-dev) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/AclManager.java:91]: I think these annotations to suppresswarnings for parameter number are unnecessary and can be removed from the refactored acls.  Ran a build without, checkstyle and findbugs, did not report a problem.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/AclManager.java [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+            volumeName + bucketName + keyName,
+            OMException.ResultCodes.PERMISSION_DENIED);
+      }
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  @SuppressWarnings("parameternumber")
```
Comment (neils-dev) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/AclManager.java:145]: Again, annotation unnecessary.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/AclManager.java [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+          ProtobufRpcEngine.Server.getRemoteIp(),
+          ProtobufRpcEngine.Server.getRemoteIp().getHostName(),
+          false, getVolumeOwner(vol, acl, resType));
+    } catch (OMException ex) {
+      // Should not trigger exception here at all
+      return false;
+    }
+  }
+
+  @SuppressWarnings("parameternumber")
```
Comment (neils-dev) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/AclManager.java:324]: annotation check, removed without issue.

**Comment by smengcl** [comment]
> Thanks @myskov for working on this.
> 
> Since there are already conflicts in `OzoneManager`, would you mind holding this off for ~7 days until snapshot feature branch is merged to master? After that I could help you on rebasing the patch and resolving the conflicts. Thanks.

Hi @myskov , the snapshot feature branch is now merged. Sorry it took 2 more days than expected. It is safe to rebase the patch now. Let me know if you need any help resolving the conflicts.

**Comment by myskov** [comment]
thanks @smengcl

**Comment by DaveTeng0** [comment]
Except for some unaddressed comments from Neil, other changes look good to me!

**Comment by neils-dev** [comment]
Thanks @myskov .  Try resolving the conflicts, rebase your branch and reach out to @smengcl if problems.  I'll go over the changes after.

**Comment by adoroszlai** [comment]
> rebase your branch

Please merge, not rebase.

**Comment by ivandika3** [comment]
Hi @myskov, are you still working on this? This is a good cleanup patch to remove all the ACL related methods from `OzoneManager` which causes a lot of confusions when I was trying to understand the Ozone ACL flow.

**Comment by ivandika3** [comment]
@adoroszlai If @myskov is not available, I can help pick this up.

**Pull request closed without merging**